pkg/queryfrontend/roundtrip.go

```diff
@@ -62,16 +64,11 @@ func NewTripperWare(
 		)
 	}
 
-	if cacheConfig != nil {
-		// constSplitter will panic when splitQueryInterval is 0.
-		if splitQueryInterval == 0 {
-			return nil, errors.New("cannot create results cache middleware when split interval is 0")
-		}
-
+	if queryRange.ResultsCacheConfig != (queryrange.ResultsCacheConfig{}) {
```

Comment:
This check is still needed here, otherwise it panics when splitting queries.

```
		if splitQueryInterval == 0 {
			return nil, errors.New("cannot create results cache middleware when split interval is 0")
		}
```

Reply:
Thanks! added to the config validation logic.